### ISSUE TITLE
problem running the makefile

### ISSUE THREAD

**Issue by archived-aft-bogus-german-visa-rejectio** [ghulette/happy-example#2]
Hi I'm pretty new to Haskell. And I'm quite new to using makefile in haskell project. I tried running `ghc --make Calc.hs` but I got `Could not find module ‘Grammar’ & ‘Tokens’` error. I tried just `make` and I got `make: alex: No such file or directory` error. If I do a `cabal init, configure & build` I would also end up getting  `Could not find module ‘Grammar’ & ‘Tokens’` error.

Would appreciate if you can be more detailed on how to use the makefile to build the project :) Thanks!

**Comment by ghulette** [comment]
Hi! As explained in README.md, you need to run two commands to install the lexer generator (alex) and the parser generator (happy) before you run make. Like this, at the command line:

```
$ cabal install alex
$ cabal install happy
```

Then just do

```
$ make
```

and it should build. Does that make sense? Try it and let me know if it doesn't work.

Also, these days the preferred way to build Haskell projects is using cabal. You can look at my more recent parser example to see how that works: https://github.com/ghulette/haskell-parser-examples

**Comment by archived-aft-bogus-german-visa-rejectio** [comment]
Hi. Thanks for the help! Yup, I read the readme and that was exactly what I did before `make`. And I got into this error: `make: alex: No such file or directory`

After some digging I realized cabal installed everything to ~/.cabal/bin & that wasn't in my path. (I was running an osx, somehow it wasn't added when i got my cabal). Everything works fine after I added it to my path in .bash_profile :) And yeah, I have checked out that repo. Nice work :+1:  Thank you so much for sharing your haskell projects with the world :) Looking into your repos is a good way to learn haskell!
